packages/design-system/src/components/web-components/preactement/define.ts

```diff
@@ -207,8 +207,15 @@ function proxyEvents(props, events: IOptions['events'], CustomElement) {
  */
 function setupMutationObserver(this: CustomElement) {
   this.__mutationObserver = new MutationObserver((mutations: MutationRecord[]) => {
-    if (mutations.find((mutation: MutationRecord) => mutation.type === 'childList')) {
-      this.renderPreactComponent();
+    const childListMutations = mutations.filter(
+      (mutation: MutationRecord) => mutation.type === 'childList'
+    );
+    if (childListMutations.length) {
+      const addedNodes = childListMutations.reduce(
```

Comment:
Could we use `.flatMap()` or another method here for readability, or is `reduce` preferred for any specific reason?

Reply:
Oh whoa, I didn't know that `flatMap` existed!